File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/RunMetric.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// RunMetric Declaration
//==============================================================
type RunMetric struct {
    gorm.Model
     Name                                    string
    Value                                                            string
    TrainingRunId         *uint
    TrainingRun           *TrainingRun `gorm:"foreignKey:TrainingRunId"`
    MetricId         *uint
    Metric           *Metric `gorm:"foreignKey:MetricId"`
    DatasetId         *uint
    Dataset           *DataSet `gorm:"foreignKey:DatasetId"`

// parent associations as their child

}

File: evidence/generated-systems/django/analytics-on-django/analyticsOnDjango/models/RunMetric.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model RunMetric
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class RunMetric Declaration
#======================================================================
class RunMetric (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	value = models.CharField(max_length=64, null=True)
	trainingRun = models.ForeignKey('TrainingRun', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	metric = models.ForeignKey('Metric', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	dataset = models.ForeignKey('DataSet', on_delete=models.CASCADE, null=True, blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.value
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "RunMetric";
    
	def objectType(self):
		return "RunMetric";
